build.zig

```diff
@@ -39,7 +39,7 @@ pub fn build(b: *std.Build) void {
     // This declares intent for the library to be installed into the standard
     // location when the user invokes the "install" step (the default step when
     // running `zig build`).
-    lib.install();
+    b.installArtifact(lib);
```

Comment:
Even after making these changes, I'm still kinda struggling on using `zshuffle` as a dependency in another project. Following the steps in the readme, it always fails to find the `zkwargs` sub-dependency:

```sh
$ zig build run
zig build-exe zig-ocr-neural-network Debug native: error: the following command failed with 1 compilation errors:
/home/eric/.zvm/0.11.0/zig build-exe /home/eric/Documents/github/zig-ocr-neural-network/src/main.zig /home/eric/Documents/github/zig-ocr-neural-network/zig-cache/o/857dab03ada5bf7c3b7b739e653f5e83/libzshuffle.a --cache-dir /home/eric/Documents/github/zig-ocr-neural-network/zig-cache --global-cache-dir /home/eric/.cache/zig --name zig-ocr-neural-network --mod zshuffle::/home/eric/.cache/zig/p/1220b05c3fbbdd2451fe974c73e7b138ee9694fb74e528c67992da7ec3596737bc0f/src/main.zig --deps zshuffle --listen=-
Build Summary: 1/6 steps succeeded; 1 failed (disable with --summary none)
run transitive failure
└─ run zig-ocr-neural-network transitive failure
   ├─ zig build-exe zig-ocr-neural-network Debug native 1 errors
   └─ install transitive failure
      └─ install zig-ocr-neural-network transitive failure
         └─ zig build-exe zig-ocr-neural-network Debug native (+1 more reused dependencies)
/home/eric/.cache/zig/p/1220b05c3fbbdd2451fe974c73e7b138ee9694fb74e528c67992da7ec3596737bc0f/src/main.zig:5:25: error: no package named 'zkwargs' available within package 'root.zshuffle'
const zkwargs = @import("zkwargs");
                        ^~~~~~~~~
referenced by:
    ShuffleRtnT: /home/eric/.cache/zig/p/1220b05c3fbbdd2451fe974c73e7b138ee9694fb74e528c67992da7ec3596737bc0f/src/main.zig:63:20
    shuffle: /home/eric/.cache/zig/p/1220b05c3fbbdd2451fe974c73e7b138ee9694fb74e528c67992da7ec3596737bc0f/src/main.zig:7:63
    remaining reference traces hidden; use '-freference-trace' to see all reference traces
```


<details>
<summary><code>build.zig</code> (standard auto-generated build script with the <code>zshuffle</code> module added)</summary>

```zig
const std = @import("std");

// Although this function looks imperative, note that its job is to
// declaratively construct a build graph that will be executed by an external
// runner.
pub fn build(b: *std.Build) void {
    // Standard target options allows the person running `zig build` to choose
    // what target to build for. Here we do not override the defaults, which
    // means any target is allowed, and the default is native. Other options
    // for restricting supported target set are available.
    const target = b.standardTargetOptions(.{});

    // Standard optimization options allow the person running `zig build` to select
    // between Debug, ReleaseSafe, ReleaseFast, and ReleaseSmall. Here we do not
    // set a preferred release mode, allowing the user to decide how to optimize.
    const optimize = b.standardOptimizeOption(.{});

    const exe = b.addExecutable(.{
        .name = "zig-ocr-neural-network",
        // In this case the main source file is merely a path, however, in more
        // complicated build scripts, this could be a generated file.
        .root_source_file = .{ .path = "src/main.zig" },
        .target = target,
        .optimize = optimize,
    });

    const zshuffle_dep = b.dependency("zshuffle", .{
        .target = target,
        .optimize = optimize,
    });
    exe.addModule("zshuffle", zshuffle_dep.module("zshuffle"));
    exe.linkLibrary(zshuffle_dep.artifact("zshuffle"));

    // This declares intent for the executable to be installed into the
    // standard location when the user invokes the "install" step (the default
    // step when running `zig build`).
    b.installArtifact(exe);

    // This *creates* a Run step in the build graph, to be executed when another
    // step is evaluated that depends on it. The next line below will establish
    // such a dependency.
    const run_cmd = b.addRunArtifact(exe);

    // By making the run step depend on the install step, it will be run from the
    // installation directory rather than directly from within the cache directory.
    // This is not necessary, however, if the application depends on other installed
    // files, this ensures they will be present and in the expected location.
    run_cmd.step.dependOn(b.getInstallStep());

    // This allows the user to pass arguments to the application in the build
    // command itself, like this: `zig build run -- arg1 arg2 etc`
    if (b.args) |args| {
        run_cmd.addArgs(args);
    }

    // This creates a build step. It will be visible in the `zig build --help` menu,
    // and can be selected like this: `zig build run`
    // This will evaluate the `run` step rather than the default, which is "install".
    const run_step = b.step("run", "Run the app");
    run_step.dependOn(&run_cmd.step);

    // Creates a step for unit testing. This only builds the test executable
    // but does not run it.
    const unit_tests = b.addTest(.{
        .root_source_file = .{ .path = "src/main.zig" },
        .target = target,
        .optimize = optimize,
    });

    const run_unit_tests = b.addRunArtifact(unit_tests);

    // Similar to creating the run step earlier, this exposes a `test` step to
    // the `zig build --help` menu, providing a way for the user to request
    // running the unit tests.
    const test_step = b.step("test", "Run unit tests");
    test_step.dependOn(&run_unit_tests.step);
}
```

</details>


<details>
<summary><code>build.zig.zon</code></summary>

```zig
.{
    .name = "zig-ocr-neural-network",
    .version = "0.1.0",
    .dependencies = .{
        .zshuffle = .{
            .url = "https://github.com/MadLittleMods/zshuffle/archive/625dc8a35031fd885fb87359e19ba6bf1baca5cc.tar.gz",
            .hash = "1220b05c3fbbdd2451fe974c73e7b138ee9694fb74e528c67992da7ec3596737bc0f",
        },
    },
}
```

</details>

Reply:
> Thanks for the updates :) The current README references a version with that transitive dependency fix included, and it works for a new project on my machine.

Working great with the latest changes (Zig 0.11.0) :white_check_mark:

Good spot on the [missing `dependencies`](https://github.com/hmusgrave/zshuffle/commit/7cc5d08a8dc849b22078d4cff728e0859941af3d)! Thanks for fixing it all up and documenting :heart: